src/main/java/seedu/duke/data/timetable/Event.java

```diff
@@ -48,18 +57,77 @@ public Event(String title, LocalDate date, LocalTime time, boolean isToRemind, b
     }
 
     /**
-     * Creates an Event object with its title, dateTime, isToRemind and isRecurring provided. Used as a default
-     * constructor for user input events.
+     * Creates an Event object with its title, date, time, isToRemind, isRecurring as well as time before reminder and
+     * time unit provided.
      *
      * @param title Title of the event.
-     * @param dateTime DateTime of the event.
-     * @param isToRemind Whether the event requires a reminder.
-     * @param isRecurring Whether the event will re-occur.
+     * @param date Date of the event.
+     * @param time Time of the event.
+     * @param isToRemind Whether event is set to remind.
+     * @param isRecurring Whether event is set to re-occur.
+     * @param timePeriods Time periods to remind about this event before it occurs.
+     * @param timeUnits Time units to remind about this event before it occurs.
+     */
+    public Event(String title, LocalDate date, LocalTime time, boolean isToRemind, boolean isRecurring,
+                 ArrayList<Integer> timePeriods, ArrayList<String> timeUnits) {
+        this(title, date, time);
+        this.isToRemind = isToRemind;
+        this.isRecurring = isRecurring;
+
+        assert timePeriods.size() == timeUnits.size() : "Something is wrong with the parser! Check it out.";
```

Comment:
I think it's fine because even in the lecture notes they talk about assert as a error handling as well as a JUnit. It can be used for both. I implemented it here as an assert because if its not the same then there would be big problems with the entire remind functionality.

Reply:
+ If there is difference in the two, it would indicate issues in the parser or implementation of the code, nothing on the user, hence exception is a less viable option.